server/CGameHandler.cpp

```diff
@@ -7082,13 +7090,46 @@ void CGameHandler::handleCheatCode(std::string & cheat, PlayerColor player, cons
 		creatures.insert(std::make_pair("vcmiainur", std::make_pair(13, 5))); //5 archangels
 		creatures.insert(std::make_pair("vcmiangband", std::make_pair(66, 10))); //10 black knights
 		creatures.insert(std::make_pair("vcmiglaurung", std::make_pair(133, 5000))); //5000 crystal dragons
+		creatures.insert(std::make_pair("vcmiarchangel", std::make_pair(13, 5))); //5 archangels
+		creatures.insert(std::make_pair("vcmiblackknight", std::make_pair(66, 10))); //10 black knights
+		creatures.insert(std::make_pair("vcmicrystal", std::make_pair(133, 5000))); //5000 crystal dragons
+		creatures.insert(std::make_pair("vcmiazure", std::make_pair(132, 5000))); //5000 azure dragons
+		creatures.insert(std::make_pair("vcmifaerie", std::make_pair(134, 5000))); //5000 faerie dragons
 
 		const CCreature * creature = VLC->creh->objects.at(creatures[cheat].first);
 		for (int i = 0; i < GameConstants::ARMY_SIZE; i++)
 			if (!hero->hasStackAtSlot(SlotID(i)))
 				insertNewStack(StackLocation(hero, SlotID(i)), creature, creatures[cheat].second);
 	}
-	else if (cheat == "vcminoldor")
+	else if (boost::starts_with(cheat, "vcmiarmy"))
```

Comment:
You can use 
`VLC->modh->identifiers.getIdentifier(CModHandler::scopeGame(), "creature", identifier);`
to find creature by ID instead of direct search in creh->objects

Reply:
I agree. Using string identifier {"archangel", 5} is quite self describing and do not need comments

Reply:
Moreover we can fallback to our new cheat vcmiarmy. vcmiarchangel = vcmiarmy archangel 5. So your new cheat does not accept the amount which might be useful in some situations

Reply:
my cheat fills each slot with different amount of creatures so it is fairly easy to adjust the amount you want - does it sound good enough?